src/App.tsx

```diff
@@ -1,41 +1,129 @@
+import { User } from './types/User';
+import { client } from './utils/fetchClient';
+import { Post } from './types/Post';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [hasPostsError, setHasPostsError] = useState(false);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  useEffect(() => {
+    client.get<User[]>('/users')
+      .then((value) => setUsers(value));
+  }, []);
+
+  useEffect(() => {
+    if (!selectedUser) {
+      return;
+    }
+
+    setIsPostsLoading(true);
+    setHasPostsError(false);
+
+    client.get<Post[]>(`/posts?userId=${selectedUser.id}`)
+      .then(posts => setUserPosts(posts))
+      .catch(() => setHasPostsError(true))
+      .finally(() => setIsPostsLoading(false));
+  }, [selectedUser]);
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  selectedUser={selectedUser}
+                  users={users}
+                  onSelect={(user) => setSelectedUser(user)}
+                />
+              </div>
 
-      <div className="App__content">
-        <PostDetails />
+              <div className="block" data-cy="MainContent">
+                {(() => {
+                  if (!selectedUser) {
+                    return (
+                      <p data-cy="NoSelectedUser">
+                        No user selected
+                      </p>
+                    );
+                  }
+
+                  if (isPostsLoading) {
+                    return (
+                      <Loader />
+                    );
+                  }
```

Comment:
Move it out of jsx into common function. Fix all cases and if you have any questions, please ask them in fe_chat and tag mentors